source/custom/nidcpower_service.custom.cpp

```diff
@@ -1,5 +1,56 @@
 #include <nidcpower/nidcpower_service.h>
 
+#include <stdexcept>
+
 namespace nidcpower_grpc {
 
+struct DriverErrorException : std::runtime_error {
+  DriverErrorException(int status) : std::runtime_error("") { status_ = status; }
+  int status_ = 0;
+};
+
+void CheckStatus(int status)
+{
+  if (status != 0) {
+    throw DriverErrorException(status);
+  }
+}
+
+::grpc::Status NiDCPowerService::MeasureMultiple(::grpc::ServerContext* context, const MeasureMultipleRequest* request, MeasureMultipleResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto vi_grpc_session = request->vi();
+      ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+      ViConstString channel_name = request->channel_name().c_str();
+
+      // Getting number of channels from ParseChannelCount API
+      ViUInt32 number_of_channels;
+      CheckStatus(library_->ParseChannelCount(vi, channel_name, &number_of_channels));
+
+      // resizing voltage measurement array and getting its pointer
+      response->mutable_voltage_measurements()->Resize(number_of_channels, 0.0);
+      ViReal64* voltage_measurements = response->mutable_voltage_measurements()->mutable_data();
+
+      // resizing current measurement array and getting its pointer
+      response->mutable_current_measurements()->Resize(number_of_channels, 0.0);
+      ViReal64* current_measurements = response->mutable_current_measurements()->mutable_data();
+
+      // calling MeasureMultiple API
+      auto status = library_->MeasureMultiple(vi, channel_name, voltage_measurements, current_measurements);
+      response->set_status(status);
+      CheckStatus(status);
```

Comment:
This `CheckStatus` is not required. Irrespective of whether it is called here or not, the above line sets the status  of `library_->MeasureMultiple` call on `response` object and returns. We do `CheckStatus` on line 31 above so that if call to `ParseChannelCount` returns error, `CheckStatus` will throw exception which is caught in `catch` section below and appropriately gets set on `response` object and we don't continue with rest of the method in such cases.

Reply:
Right, got it. Done the changes.